catalog/core/catalog-core-directorymonitor/src/main/java/org/codice/ddf/catalog/content/monitor/DavAlterationObserver.java

```diff
@@ -0,0 +1,310 @@
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.catalog.content.monitor;
+
+import java.io.IOException;
+import java.io.Serializable;
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.List;
+import java.util.concurrent.CopyOnWriteArrayList;
+
+import com.github.sardine.DavResource;
+import com.github.sardine.Sardine;
+
+/**
+ * A webdav implementation of the {@link org.apache.commons.io.monitor.FileAlterationObserver}
+ * Uses https://github.com/lookfirst/sardine
+ */
+public class DavAlterationObserver implements Serializable {
+
+    private static final long serialVersionUID = 42L;
+
+    private static final DavResource[] EMPTY_RESOURCES = new DavResource[] {};
+
+    private transient List<EntryAlterationListener> listeners = new CopyOnWriteArrayList<>();
+
+    private final DavEntry rootEntry;
+
+    private transient Sardine sardine;
+
+    DavAlterationObserver(final DavEntry rootEntry) {
+        if (rootEntry == null) {
+            throw new IllegalArgumentException("Root entry is missing");
+        }
+        this.rootEntry = rootEntry;
+    }
+
+    /**
+     * Return the directory being observed.
+     *
+     * @return the directory being observed
+     */
+    public String getDirectory() {
+        return rootEntry.getLocation();
+    }
+
+    /**
+     * Add a file system listener.
+     *
+     * @param listener The file system listener
+     */
+    void addListener(final EntryAlterationListener listener) {
+        if (listener != null) {
+            listeners.add(listener);
+        }
+    }
+
+    /**
+     * Remove a file system listener.
+     *
+     * @param listener The file system listener
+     */
+    void removeListener(final EntryAlterationListener listener) {
+        if (listener != null) {
+            while (listeners.remove(listener)) {
+            }
+        }
+    }
+
+    /**
+     * Returns the set of registered file system listeners.
+     *
+     * @return The file system listeners
+     */
+    public Iterable<EntryAlterationListener> getListeners() {
+        return listeners;
+    }
+
+    /**
+     * Initialize the observer. Note this means no events will be fired for
+     * files that already existed before monitoring started.
+     *
+     * @param sardine
+     */
+    public void initialize(Sardine sardine) {
+        this.sardine = sardine;
+        try {
+            DavResource davResource = sardine.list(rootEntry.getLocation())
+                    .get(0);
+            rootEntry.refresh(davResource);
+            final DavEntry[] children = doListFiles(rootEntry.getLocation(), rootEntry);
+            rootEntry.setChildren(children);
+        } catch (IOException e) {
+            // probably means root location was inaccessible
+            throw new RuntimeException(e);
+        }
+    }
+
+    /**
+     * Final processing.
+     *
+     * @throws Exception if an error occurs
+     */
+    public void destroy() throws Exception {
```

Comment:
❓ why do you have this method if does not do anything? I see it is being called by DurableWebDavFileConsumer, but since it is calling this implementation directly should something actually happen or does DurableWebDavFileConsumer not really need to call this method?